### PR TITLE
add support for detecting auto-downloaded dependencies in PythonPackage easyblock

### PR DESCRIPTION
**Pull request by boegel**
edit: https://github.com/easybuilders/easybuild-framework/pull/2493 + depends on https://github.com/easybuilders/easybuild-framework/pull/2494

### CODE REVIEW

File: easybuild/easyblocks/generic/pythonpackage.py [hunk 1]
```diff
@@ -178,17 +178,18 @@ def extra_options(extra_vars=None):
         if extra_vars is None:
             extra_vars = {}
         extra_vars.update({
-            'unpack_sources': [True, "Unpack sources prior to build/install", CUSTOM],
+            'buildcmd': ['build', "Command to pass to setup.py to build the extension", CUSTOM],
+            'download_dep_fail': [False, "Fail if downloaded dependencies are detected", CUSTOM],
```
Comment (boegel) [inline easybuild/easyblocks/generic/pythonpackage.py]: Set to `False` for backward-compatibility, but we should consider switching this to `True` by default for EasyBuild v4.0.0.

Problem is that we'll need to fix quite a bit of easyconfigs...

We could try to enforce a policy where new easyconfigs for Python packages include `download_dep_fail = True`?

**Comment by boegel** [comment]
@wpoely86 please review?

**Comment by boegel** [comment]
@akesandgren suggestion is now implemented, but it depends on two changes in EasyBuild framework (see updated description)

**Comment by boegel** [comment]
The framework PRs on which this depends have been merged, so this is good to go?

**Comment by vanzod** [comment]
My tests did not show any issue. Merging

**Pull request merged**